gamersnet_backend/__tests__/posts/testPostUpdate.js

```diff
@@ -0,0 +1,160 @@
+'use strict';
+
+let app = require('../../app');
+let request = require('supertest');
+
+let MongoDB = require('../../persistence/mongodb');
+let db;
+const { ObjectID, ObjectId } = require('bson');
+
+let users, tokens, posts;
+let user1ID, user2ID;
+let post1ID, post2ID;
+
+let user1 = {username: "user1", password: "123"}
+let user2 = {username: "user2", password: "456"}
+
+let token1 = {token: "user1_token", userID: ObjectID(), expires: 9999999999999}//logged in
+let token2 = {token: "user2_token", userID: ObjectID(), expires: 0}//logged in
+
+let post1 = {
+    userID : ObjectID(),
+    description: "post by user1",
+    gameName: "xyz",
+    numPlayers: 5,
+    gameTimeUTC: new Date(),
+    duration: "1hr",
+    location: "Earth"
+}
+let post2 = {
+    userID : ObjectID(),
+    description: "post by user2",
+    gameName: "xyz",
+    numPlayers: 5,
+    gameTimeUTC: new Date(),
+    duration: "1hr",
+    location: "Pluto"
+}
+
+beforeAll(async () => {
+    if (db) await MongoDB.close();
+    db = await MongoDB.open();
+
+    jest.setTimeout(10000);
+    //create collections
+    users = db.collection("users");
+    tokens = db.collection("tokens");
+    posts = db.collection("posts");
+
+    //set up DB with mock data
+    await seedDB();
+});
+
+async function seedDB() {
+
+    //insert users
+    await users.deleteMany();
```

Comment:
Is this deleting all users' data? Or just the ones that are being inserted in any test?

Reply:
The ones in this test. I don't think we need it since the db is empty anyway, but i put it there just in case

Reply:
Even if it is for the test, we need it because seedDB() is getting called before each test is being run and we want to make sure tests are independent of any data inserted/deleted.

Reply:
The following lines (which are in the `beforeAll` function call) will stop the MongoDB server running on the local machine and start it again, therefore clearing all data. It also appears that this is one of this things that could have been coded better, but I coded it and never looked back